src/locutus/model/terminology.py

```diff
@@ -227,22 +231,29 @@ def rename_code(
                         self.set_mapping(new_code, mappings[original_code])
                         self.delete_mappings(code=original_code, editor=editor)
 
-                if new_display is not None:
+                if new_display is not None and code.display != new_display:
+                    old_values.append(f"display: {code.display}")
+                    new_values.append(f"display: {new_display}")
                     code.display = new_display
 
-                if new_description is not None:
+                if new_description is not None and code.description != new_description:
+                    old_values.append(f"description: {code.description}")
+                    new_values.append(f"description: {new_description}")
                     code.description = new_description
 
+                old_values = ",".join(old_values)
+                new_values = ",".join(new_values)
+
                 self.save()
-                # avoid using add_provenance if code did NOT change
-                self.add_provenance(
-                    change_type=Terminology.ChangeType.EditTerm,
-                    target=original_code,
-                    old_value=original_code,
-                    new_value=new_code,
-                    editor=editor,
-                )
-                return True
+                if old_values and new_values:

```

Comment:
I imagine only one of these is necessary for this conditional. Each evaluation inside that and is a separate set of machine instructions and take up CPU time. If you can eliminate one of them, you will save a bit of effort for the machines it runs on. And, given the potential frequency with which this call may be used, it seems worth considering.